pf/internal/schemashim/object_pseudoresource.go

```diff
@@ -94,24 +120,34 @@ func (r *objectPseudoResource) Get(key string) shim.Schema {
 }
 
 func (r *objectPseudoResource) GetOk(key string) (shim.Schema, bool) {
-	if attr, ok := r.attrs[key]; ok {
+	// There is something here that possibly could be done better.
+	//
+	// When moving down a property identified by key, we are interested in keeping track of Attr for that property,
+	// and recurse using attrSchema. This information may be coming out of band from the ObjectTypeable value
+	// itself when using blocks, see TestCustomTypeEmbeddingObjectType.
+	if attr, ok := r.nestedAttrs[key]; ok {
 		return &attrSchema{key, attr}, true
 	}
 
-	if t, ok := r.obj.AttrTypes[key]; ok {
-		return newTypeSchema(t, nil), true
+	// If there fail to find an Attr, perhaps we have a simple ObjectType then we can look up the property's
+	// AttrType and go with that.
+	if objType, ok := r.obj.(types.ObjectType); ok {
+		if t, ok := objType.AttrTypes[key]; ok {
+			return newTypeSchema(t, nil), true
+		}
+	}
+
+	for _, a := range r.allAttrNames {
+		if key == a {
+			panic("[pf/internal/schemashim] Attribute %q should be defied was not found, this should never happen")
```

Comment:
Does the `%q` refer to anything?

Again, I would convert this to a `contract.Assertf(key != a, "...` interface.